app/models/mixins/miq_request_mixin.rb

```diff
@@ -169,4 +169,13 @@ def request_dialog(action_name)
     dialog = ResourceActionWorkflow.new(values, get_user, ra, {}).dialog
     DialogSerializer.new.serialize(Array[dialog]).first
   end
+
+  def set_option(key, value)
+    current_class = options[key.to_sym].class
+    if value.class != current_class && current_class != NilClass
```

Comment:
Since we do not control all the keys stored here, some will be user defined values, this may be to restrictive.  I am thinking users should also be able to change a value to `nil`.

Reply:
Rethinking this over the weekend and I realized all this does is create a work-around to what this logic is trying to enforce.  Meaning, if someone wanted to change a type of a field they would end up just setting to `nil` first and then could set anything they wanted.  So I withdraw my previous comment.  😞